integration/tests/cook/test_dynamic_clusters.py

```diff
@@ -0,0 +1,107 @@
+                                lambda x: True if 201 == x[1].status_code else x[1].content,
+                                300000, 5000)
+            # Create at least one new cluster with a unique test name (using one of the existing cluster's IP and cert)
+            test_cluster_name = f'test_cluster_{round(time.time() * 1000)}'
+            test_cluster = {
+                "name": test_cluster_name,
+                "state": "running",
+                "base-path": clusters[0]["base-path"],
+                "ca-cert": clusters[0]["ca-cert"],
+                "template": clusters[0]["template"]
+            }
+            data, resp = util.create_compute_cluster(self.cook_url, test_cluster)
+            self.assertEqual(201, resp.status_code, resp.content)
+
+        # Check that a job schedules successfully
+        command = "true"
+        job_uuid, resp = util.submit_job(self.cook_url, command=command, container=container)
+        self.assertEqual(201, resp.status_code, resp.content)
+        util.wait_for_instance(self.cook_url, job_uuid, status='success')
+
+        with admin:
+            # Delete test cluster
+            # First state = draining
+            test_cluster["state"] = "draining"
+            data, resp = util.update_compute_cluster(self.cook_url, test_cluster)
+            self.assertEqual(201, resp.status_code, resp.content)
+            # Then state = deleted
+            test_cluster["state"] = "deleted"
+            util.wait_until(lambda: util.update_compute_cluster(self.cook_url, test_cluster),
+                            lambda x: True if 201 == x[1].status_code else x[1].content,
+                            300000, 5000)
+            # Hard-delete the original non-test clusters
+            for cluster in clusters:
+                self.logger.info(f'Trying to delete cluster {cluster}')
+                resp = util.delete_compute_cluster(self.cook_url, cluster)
+                self.assertEqual(204, resp.status_code, resp.content)
+            # Force give up leadership
+            self.logger.info(f"COOK_CHANGE_LEADER_COMMAND {os.getenv('COOK_CHANGE_LEADER_COMMAND')}")
+            cp = subprocess.run(os.getenv('COOK_CHANGE_LEADER_COMMAND'), shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+            self.assertEqual(0, cp.returncode, f'{cp.stdout} {cp.stderr}')
+
+        # Old clusters should be re-created
+        # wait for cook to come up
+        util.wait_until(lambda: [cluster for cluster in util.compute_clusters(self.cook_url)['db-configs'] if cluster["state"] == "running"],
+                        lambda x: len(x) == len(clusters),
+                        420000, 5000)
+        # Check that a job schedules successfully
+        command = "true"
+        job_uuid, resp = util.submit_job(self.cook_url, command=command, container=container)
+        self.assertEqual(201, resp.status_code, resp.content)
+        util.wait_for_instance(self.cook_url, job_uuid, status='success')
```

Comment:
Should we submit jobs until each of the old clusters has successfully run a job?

Reply:
Why is it fine if one of them isn't working? What's the downside to checking that both are working?

Reply:
there's no reason to think that one will work but another won't
But if we want to test them all, how we get a job to schedule on each one?

Reply:
I think you'd have to submit jobs until one of them gets matched to each compute cluster. Let me know what you think. If you think it's not worth the complexity, I'm fine with it.

Reply:
we won't test here, we'll do it as a follow up test